### PR TITLE
Include relay/crank balance checker in scheduled payments data check

### PR DESCRIPTION
**Pull request by jurgenwerk**
This PR adds some basic monitoring to see if our relayer and the crank have enough balance to fund scheduling tool transactions (scheduling and executing). If the balance is too low to pay for transactions gas, the scheduling or executing will stop working, so we need to make sure we are informed when topping up is needed. 

We are running a Checkly check that continuously makes requests to our data integrity checks endpoint, which includes scheduled payments check. This PR adds the balance check. If the check discovers the balance is low, we will get alarmed in our discord channels and also in PagerDuty (for production). 

The low balance threshold is currently set at 0.05 of native coin value (e.g ETH, MATIC). There will also be a process described where a person in charge will manually transfer the funds from the fee receiver to the crank/relayer when this automatic check will trigger an alarm. 

**To discuss** (I am working through these questions now but comments welcome)

Is 0.05 a good threshold for both relayer and the crank? Does that vary on different networks significantly? Ideally the threshold should be at least a multiplication of the gas cost. Would it be good to introduce separate thresholds for relayer and the crank?

### CODE REVIEW

File: packages/hub/services/data-integrity-checks/scheduled-payments.ts [hunk 1]
```diff
@@ -10,6 +14,15 @@ export const UNATTEMPTED_ALLOWED_MINUTES = 24 * 60;
 export const ATTEMPTS_WITHOUT_TX_HASH_ALLOWED_MINUTES = 60;
 export const UNFINISHED_ATTEMPT_DURATION_ALLOWED_MINUTES = 24 * 60;
 
+// Relayer needs funds to pay for safe transactions gas (e.g creating safes, scheduling a payment).
```
Comment (jurgenwerk) [inline packages/hub/services/data-integrity-checks/scheduled-payments.ts:17]: Note to self: remove "creating safes" – that actually comes from the user's wallet!

File: packages/hub/services/data-integrity-checks/scheduled-payments.ts [hunk 2]
```diff
@@ -10,6 +14,15 @@ export const UNATTEMPTED_ALLOWED_MINUTES = 24 * 60;
...
+// Gas cost reimbursement for the relayer will be paid from the (specified) gas token balance in the user's safe.
+const LOW_RELAYER_FUNDS_THRESHOLD = ethers.utils.parseEther('0.05');
+
+// Crank (i.e the hub) needs funds to pay for scheduled payment execution transactions gas.
+// This gets reimbursed to the fee receiver using platform defined fees collected from the
+// scheduled payment (for example $0.25 USD and 0.1% of the transaction amount)
```
Comment (FadhlanR) [inline packages/hub/services/data-integrity-checks/scheduled-payments.ts:23]: And also gets gas reimbursement for the execution process.

**Review by FadhlanR — APPROVED** [review]

**Comment by jurgenwerk** [comment]
Per our discussion yesterday we decided a good threshold for now would be something like expected users (500) * 50 transactions/month * avg transaction gas cost. 

However this sounds like a huge amount for networks with high gas fees, for example Ethereum. In that case the avg gas fee per tx has been around [0.6 USD](https://ycharts.com/indicators/ethereum_average_transaction_fee#:~:text=Ethereum%20Average%20Transaction%20Fee%20is,61.40%25%20from%20one%20year%20ago) which is around 0.0003 ETH (avg gas fee for an Ethereum transaction). If we calculate the threshold, we get: 500 * 50 * 0.0003 = 7.5 ETH. 

In case of Polygon, the average gas fee per tx seems to be around 0.03 MATIC. Plugging that into the formula we get 500 * 50 * 0.03 = 750 MATIC

I think maintaining ~7.5 ETH and ~750 MATIC balance for both crank and relayer sounds quite a lot for now when the user activity is still very low. 

I would suggest to simply hardcode values for supported network and revisit them later if scaling is needed.

1. Ethereum (mainnet and goerli): 0.3 ETH 
2. Polygon: 30 MATIC 

This would still suffice for ~1000 transactions once we get a warning.

File: packages/hub/services/data-integrity-checks/scheduled-payments.ts [hunk 3]
```diff
@@ -152,12 +166,53 @@ export default class DataIntegrityChecksScheduledPayments {
...
+        );
+      }
+    }
+  }
+
+  async getRelayerFunderBalance(networkName: string): Promise<BigNumber> {
+    let relayerUrl = getConstantByNetwork('relayServiceURL', networkName);
+
+    let responseText = await (await fetch(`${relayerUrl}/v1/about`)).text();
+    let relayerFunderPublicKey = JSON.parse(responseText).settings.SAFE_FUNDER_PUBLIC_KEY;
```
Comment (jurgenwerk) [inline packages/hub/services/data-integrity-checks/scheduled-payments.ts:205]: Note to self: SAFE_FUNDER_PUBLIC_KEY is wrong, SAFE_TX_SENDER_PUBLIC_KEY is correct

**Pull request merged**